### DISCUSSION TITLE
Hi! Do we have now ability to order by the one-to-many relation in both direction ?

### DISCUSSION THREAD

**Discussion by mralexfed007** [prisma/prisma#22954] [Q&A] [answered]
> Hey folks, we started working on this and are ready to share our proposal, please have a read and let us know if we're missing anything.
> 
> If interested, I'd love to jump on a short call with you to chat about this proposal or any problems you face with the Prisma Client: https://calendly.com/matthewmueller/prisma. This is a great opportunity to influence what we build!
> 
> ---
> 
> # Proposal
> 
> ## Databases
> 
> - Postgres
> - MySQL
> - SQLite
> - SQL Server
> 
> ## Operations
> 
> - findMany
> - findFirst
> - groupBy (maybe)
> 
> ## Order Properties
> 
> - ascending (asc)
> - descending (desc)
> 
> ## Relations
> 
> Ordering by relations is supported on both one-to-one and one-to-many relations. In both cases, you can traverse the relation on the side of the relation that "has one". Let's look at some examples
> 
> ### One-to-One
> 
> For one-to-one relationships, you can order by the relation from either side:
> 
> ✅ Order users by their budget if they're a customer
> 
> ✅ Order customers by by their budget if they're a customer
> 
> ```groovy
> model User {
>   id Int @id
>   name String
>   customer Customer?
> }
> 
> model Customer {
>   id Int @id
>   budget Int
>   user User
> }
> ```
> 
> ```tsx
> prisma.user.findMany({
>   orderBy: [{
>     customer: {
>       budget: 'desc'
>     }
>   }]
> })
> ```
> 
> ### One-to-Many
> 
> For one-to-many relationships, you can order by the relation from one side:
> 
> ✅ Order posts by their author's name
> 
> ⛔️ Order authors by their post's titles
> 
> ```groovy
> model Author {
>   id Int @id
>   name String
>   posts Post[]
> }
> 
> model Post {
>   id Int @id
>   title String
>   author Author
> }
> ```
> 
> ```tsx
> prisma.post.findMany({
>   orderBy: [{
>     author: {
>       name: 'asc'
>     }
>   }]
> })
> ```
> 
> # FAQ
> 
> ## Can you order by multiple fields inside a relation?
> 
> Yes! Just like with ordering by multiple scalars, you can also order by multiple relation scalars. 
> 
> ```tsx
> prisma.post.findMany({
>   select: {
>     author: {
>       firstName: true // will issue a 2nd, unnecessary SELECT query
>     } 
>   },
>   orderBy: [{
>     author: {
>       firstName: 'asc'
>     },
>     author: {
> 	  lastName: 'desc'
>     }
>   }]
> })
> ```
> 
> This somewhat clumsy syntax is necessary because the order matters for fields in an `orderBy`.
> 
> ## Can you order by a deeply nested relation?
> 
> Surprisingly, yes. You can also order by a deeply nested relation. Given the following schema:
> 
> ```prisma
> model Author {
>   id Int @id
>   name String
>   posts Post[]
>   address Address
> }
> 
> model Post {
>   id Int @id
>   title String
>   author Author
> }
> 
> model Address {
>   id Int @id
>   title String
>   author Author
> }
> ```
> 
> ```tsx
> prisma.post.findMany({
>   orderBy: [{
>     author: {
>       firstName: 'asc',
>     },
>     author: {
>       address: {
>         zip: 'desc'
>       }
>     }
>   }]
> })
> ```
> 
> ## Can you order by a foreign key?
> 
> Yep. 
> 
> ```tsx
> await prisma.post.findMany({
>   orderBy: [{
>     author: {
>       addressId: 'asc'
>     }
>   }]
> })
> ```
> 
> ## How do you handle NULLs?
> 
> Just like with ordering by scalars, how NULLs are ordered in a column is database-specific. We don't impose any additional ordering on top of what the database already does:
> 
> ```tsx
> await prisma.post.findMany({
>   orderBy: [{
>     author: {
>       name: 'asc' // handling NULL names is database-specific
>     },
>   }]
> })
> ```
> 
> # Limitations
> 
> ## Unnecessary SELECT while doing a nested select
> 
> The following will issue a JOIN and a SELECT when the JOIN has the record's information already
> 
> ```tsx
> prisma.post.findMany({
>   select: {
>     author: {
> 	  firstName: true // will issue a 2nd, unnecessary SELECT query
>     } 
>   },
>   orderBy: [{
>     author: {
>       firstName: 'asc'
>     }
>   }]
> })
> ```
> 
> We'll address this later when we focus on query optimization.
> 
> _Originally posted by @matthewmueller in https://github.com/prisma/prisma/issues/5008#issuecomment-768444693_
>

**Comment by mralexfed007** [comment]
I mean can we make Order authors by their post's titles described in "One-to-Many" section ?

**Comment by ludralph** [comment, marked as answer]
Hi @mralexfed007 👋 

Thank you for raising this question.

It appears that for one-to-many relationships, you can order by the relation from one side. For example, you can order posts by their author's name. However, the reverse, ordering authors by their post's titles, is not yet supported.

However, we do have a [feature request](https://github.com/prisma/prisma/issues/5837) for it. You can add your use case and leave a 👍 to help our engineers prioritize it. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by Verow23** [comment]
Is there also in issue for ordering a many-to-many relation in both directions?
